numpy/core/arrayprint.py

```diff
@@ -880,9 +893,10 @@ def fillFormat(self, data):
                 self.unique = True
                 self.trim = '.'
 
-        # account for sign = ' ' by adding one to pad_left
-        if all(finite_vals >= 0) and self.sign == ' ':
-            self.pad_left += 1
+        if self._legacy != '1.13':
+            # account for sign = ' ' by adding one to pad_left
+            if not any(np.signbit(finite_vals)) and self.sign == ' ':
```

Comment:
Is there a test for signed zero? Nice catch!

Reply:
Yup, I added the following test in lines 444 and 452 of test_arrayprint:
```python
assert_equal(repr(np.array([0., -0.])), 'array([ 0., -0.])')
```